tests/state/test_get_states.py

```diff
@@ -0,0 +1,365 @@
+        '--address',
+        ADDRESS_WITH_STATE,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_address = json.loads(result.output).get('result')[0].get('address')
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert expected_address == ADDRESS_WITH_STATE
+
+
+@pytest.mark.parametrize('flag', ['address', 'start'])
+def test_get_states_with_invalid_address_start(flag):
+    """
+    Case: get a list of states by its invalid address and account address starting from.
+    Expect: the following address is invalid error message.
+    """
+    invalid_address = '044c7db163cf2'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--' + flag,
+        invalid_address,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': {
+            flag: [
+                f'The following address `{invalid_address}` is invalid.',
+            ],
+        },
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+@pytest.mark.parametrize('flag', ['--address', '--start'])
+def test_get_states_with_non_existing_address_start(flag):
+    """
+    Case: get a list of states by its non-existing address and account address starting from.
+    Expect: block not found error message.
+    """
+    non_existing_address = '0000000000000000000000000000000000000000000000000000000000100000000031'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        flag,
+        non_existing_address,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': f'Block not found.',
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+def test_get_states_with_start():
+    """
+    Case: get a list of states by account address starting from.
+    Expect: list of states is returned.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--start',
+        ADDRESS_WITH_STATE,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    result_data = json.loads(result.output).get('result')[0]
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert result_data
+
+
+def test_get_states_with_limit():
+    """
+    Case: get a list of states by limit.
+    Expect: list of states is returned.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--limit',
+        1,
+        '--node-url',
+        DEV_BRANCH_NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    result_data = json.loads(result.output).get('result')[0]
```

Comment:
Again, create more meaningful tests. If this is limit — check the length.

```
limit = 1
...

result_states = json.loads(result.output).get('result')
assert len(result_states) == limit
```

Reply:
@dmytrostriletskyi, done.